shell/common/renderer_context_manager.h

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_RENDERER_CONTEXT_MANAGER_H_
+#define FLUTTER_SHELL_COMMON_RENDERER_CONTEXT_MANAGER_H_
+
+#include <functional>
+#include <memory>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/macros.h"
+
+namespace flutter {
+
+template <typename RendererContext>
```

Comment:
It is unclear why this has to be templated. The render context can be an abstract class instead that the implementation can subclass. This will also allow you to OOL these methods.

Reply:
I was wondering how to test this, as the implementation in objc (with a EAGLContext) wasn't testable ASAIK. So I made this change so at least we can test the flow with some other type.

Reply:
If RenderContext was an abstract base class instead, that would solve your testability problem

Reply:
discussed offline: 
```excessive use of templating quickly leads to code bloat and binary size increases. If the only use case is testability, the abstract base class makes more sense.```
I'm going to update this PR to make each platform override this.